### PR TITLE
Ensure Nessus instance has access to STS and SSM endpoints before launch

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##
This PR adds two dependencies to the Nessus instance for the security group rules that allow access to the STS and SSM endpoints.

## 💭 Motivation and context ##
While redeploying one of our environments, I observed a failure of [our Nessus setup script](https://github.com/cisagov/cool-assessment-terraform/blob/develop/cloud-init/nessus-setup.sh):

```
Cloud-init v. 20.2 running 'modules:config' at Mon, 08 Feb 2021 19:51:54 +0000. Up 75.12 seconds.
Assuming role that can read Nessus-related SSM Parameter Store parameters

Could not connect to the endpoint URL: "https://sts.us-east-1.amazonaws.com/"
Reading Nessus-related parameters from SSM Parameter Store...

Could not connect to the endpoint URL: "https://ssm.us-east-1.amazonaws.com/"
WARNING: Could not read admin username from SSM!

Could not connect to the endpoint URL: "https://ssm.us-east-1.amazonaws.com/"
WARNING: Could not read admin password from SSM!
```

I believe this happened because Terraform launched the Nessus instance before it created the security group rules that allow access to the STS and SSM endpoints.  This PR attempts to correct that situation.

## 🧪 Testing ##
I confirmed that `terraform apply` ran without error (no changes as a result of this PR), but other than that, I am just going to have to keep an eye out to see if this issue arises again and if so, do further investigation.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
Makes sense ✔

File: nessus_ec2.tf [hunk 1]
```diff
@@ -33,6 +33,15 @@ resource "aws_instance" "nessus" {
   count    = lookup(var.operations_instance_counts, "nessus", 0)
   provider = aws.provisionassessment
 
+  # When a Nessus instance starts up, it executes cloud-init/nessus-setup.sh,
+  # which requires access to the STS and SSM endpoints.  To ensure that access
+  # is available, we force dependencies on the security group rules that
+  # allow STS and SSM endpoint access from Nessus.
+  depends_on = [
+    aws_security_group_rule.ingress_from_nessus_to_sts_via_https,
+    aws_security_group_rule.ingress_from_nessus_to_ssm_via_https
```
Comment (jsf9k) [inline nessus_ec2.tf:42]: You may want to list the VPC endpoints themselves as well.  They are secretly EC2 instances under the hood, and they take some time to create.

Sorry I keep having these thoughts _after_ the PR has been approved.

Comment (dav3r) [reply nessus_ec2.tf:42]: Just to be clear, are you talking about adding dependencies on `aws_vpc_endpoint.sts` and `aws_vpc_endpoint.ssm` here?

https://github.com/cisagov/cool-assessment-terraform/blob/1896e889886d71aaf0f1c834935fa927fac201c6/vpc_endpoints.tf#L9-L35

Also, do I need to include `aws_vpc_endpoint.ssmmessages` as well?

https://github.com/cisagov/cool-assessment-terraform/blob/1896e889886d71aaf0f1c834935fa927fac201c6/vpc_endpoints.tf#L76-L87

Comment (jsf9k) [reply nessus_ec2.tf:42]: Yes, I'm talking about adding dependencies on `aws_vpc_endpoint.sts` and `aws_vpc_endpoint.ssm`. 
Does cloud-init need access to any services after SSM Parameter Store, such S3?  If so then those should be listed as well.

`aws_vpc_endpoint.ec2`, `aws_vpc_endpoint.ec2messages`, `aws_vpc_endpoint.kms`, and `aws_vpc_endpoint.ssmmessages` are all used by the Amazon AWS SSM agent, but should not be necessary for accessing SSM Parameter Store; hence, there is no need to list those endpoints in the `depends_on`.

Comment (dav3r) [reply nessus_ec2.tf:42]: The only cloud-init in this repo that needs S3 is for the Guacamole instance to fetch it's SSL certificate (see [install-certificates.py](https://github.com/cisagov/cool-assessment-terraform/blob/develop/cloud-init/install-certificates.py)).  I can add a `depends_on` to the Guacamole instance for `aws_vpc_endpoint.s3` as well.

Comment (dav3r) [reply nessus_ec2.tf:42]: This has been taken care of in #103 - thanks @jsf9k for the suggestions!

**Pull request merged**